Packs/OPSWAT-Filescan/Integrations/OPSWATFilescan/OPSWATFilescan.py

```diff
@@ -0,0 +1,390 @@
+        elif verdict.upper() == "SUSPICIOUS":
+            score = Common.DBotScore.SUSPICIOUS
+        else:
+            score = Common.DBotScore.NONE
+
+        analysis_file = analysis.get("file", {})
+        dbot_score = Common.DBotScore(
+            indicator=analysis_file.get("sha256"),
+            indicator_type=DBotScoreType.FILE,
+            integration_name="OPSWAT Filescan",
+            score=score,
+        )
+
+        file = Common.File(
+            name=analysis_file.get("name"),
+            sha256=analysis_file.get("sha256"),
+            dbot_score=dbot_score,
+        )
+
+        headers = ["File Name", "SHA256", "Verdict"]
+        hr = {
+            "File Name": analysis_file.get("name"),
+            "SHA256": analysis_file.get("sha256"),
+            "Verdict": verdict,
+        }
+        results = CommandResults(
+            outputs_prefix=f"{INTEGRATION_CONTEXT_NAME}.Analysis",
+            outputs_key_field="sha256",
+            outputs=hr,
+            readable_output=tableToMarkdown("File Result:", hr, headers),
+            indicator=file,
+        )
+        return results
+
+    hr_headers = [
+        "Id",
+        "SampleName",
+        "SHA256",
+        "Verdict",
+        "State",
+        "Date",
+        "MIMEType",
+        "Type",
+        "Tags",
+    ]
+
+    command_res_ls = []
+    hr_analysis_ls = []
+
+    analyses = raw_result.get("items", {})
+
+    for analysis in analyses:
+        hr_analysis_ls.append(build_analysis_hr(analysis))
+        command_res_ls.append(build_indicator_object(analysis))
+
+    command_result = CommandResults(
+        outputs=analyses,
+        readable_output=tableToMarkdown("Analysis Result:", hr_analysis_ls, hr_headers),
+        outputs_prefix=f"{INTEGRATION_CONTEXT_NAME}.Analysis",
+    )
+
+    command_res_ls.append(command_result)
+    return command_res_ls
+
+
+def sample_submission(client: Client, args: Dict[str, Any]) -> PollResult:
+    res = client.post_sample(args)
+    partial_res = CommandResults(
+        readable_output=f'Waiting for submission "{res.get("flow_id")}" to finish...'
+    )
+    return PollResult(
+        response=CommandResults(
+            outputs=res,
+            outputs_prefix=f"{INTEGRATION_CONTEXT_NAME}.Submission",
+            outputs_key_field="flow_id",
+        ),
+        args_for_next_run={"flow_id": res.get("flow_id"), **args},
+        continue_to_poll=True,
+        partial_result=partial_res,
+    )
+
+
+def build_reputation_result(api_reponse: Dict[str, Any]):
+    reports = api_reponse.get("reports", {})
+
+    if len(reports) == 1:
+        first_report = next(iter((api_reponse.get("reports", {}).items())))[1]
+        return build_one_reputation_result(first_report)
+
+    elif len(reports) > 1:
+        command_res_ls = []
+        for report in reports:
+            command_res_ls.append(build_one_reputation_result(reports[report]))
+        return command_res_ls
```

Comment:
```suggestion
def build_reputation_result(api_reponse: Dict[str, Any]):
    reports = api_reponse.get("reports", {})
    if isinstance(reports, dict):  # there is only a single report
        reports = [reports]
    
    command_res_ls = []
    for report in reports:
        command_res_ls.append(build_one_reputation_result(reports[report]))
     return command_res_ls
```

maybe something like that?

Reply:
Actually, `reports` is already a list, so it doesn't necessary to put it to another list like this: reports = [reports]
I simplified the code and corrected the relevant test case as well.

Reply:
in that case make sure the default value will be `list` if it does not exist:
`reports = api_reponse.get("reports", [])`

Reply:
Corrected!